include/xtensor/xbuffer_adaptor.hpp

```diff
@@ -767,6 +795,15 @@ namespace xt
             swap(m_size, rhs.m_size);
             swap(m_allocator, rhs.m_allocator);
         }
+
+        template <class CP, class A>
+        template <class P>
+        inline void xbuffer_owner_storage<CP, A>::replace(P&& data, size_type size, const allocator_type& alloc) noexcept
+        {
+            m_data = data;
```

Comment:
That's a memory leak since `xbuffer_owner_storage` is responsible for deleting its data. A more robust implementation would be to use the copy and swap idiom.